releasenotes/notes/exception-when-program-runs-too-long-cf80743eda83d601.yaml

```
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    Raise an exception when a job runs for too long so that it can be handled appropriately by programs.
```

@kwoane this description can be a bit more specific since it will go on the release notes page here https://qiskit.org/documentation/partners/qiskit_ibm_runtime/release_notes.html and people should not have to read the code changes to understand it.

Something like:
```suggestion
    Raise ``RuntimeJobMaxTimeoutError`` when a job runs for too long so that it can be handled appropriately by programs.
```